middleware/session/middleware.go

```diff
@@ -0,0 +1,304 @@
+// Package session provides session management middleware for Fiber.
+// This middleware handles user sessions, including storing session data in the store.
+package session
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/gofiber/fiber/v3"
+	"github.com/gofiber/fiber/v3/log"
+)
+
+// Middleware holds session data and configuration.
+type Middleware struct {
+	Session   *Session
+	ctx       *fiber.Ctx
```

Comment:
is pointer necessary here?i think the interface itself can be treated as a pointer

Reply:
Thanks for the feedback! You're right that interfaces are often treated like pointers. However, in this case, using `*fiber.Ctx` was intentional. AFAIK Since `fiber.Ctx` is a struct, passing it by value would create copies, which could lead to unnecessary overhead and inconsistent state. Using a pointer ensures we're working with the same instance, especially important when reusing objects from `sync.Pool`.

Let me know if you think there's a better way! Happy to discuss further.

Reply:
Copilot says:

```
Using *fiber.Ctx is better in this context. Here’s why:

Performance: Passing a pointer (*fiber.Ctx) is more efficient than passing a value (fiber.Ctx) because it avoids copying the entire fiber.Ctx struct, which can be large.
Mutability: Using a pointer allows the middleware to modify the fiber.Ctx object directly, which is often necessary in middleware to set response headers, status codes, etc.
Consistency: Fiber's context (fiber.Ctx) is designed to be used as a pointer throughout the framework, ensuring consistency and expected behavior.
Therefore, the current usage of ctx *fiber.Ctx is appropriate.
```

Reply:
Considering the changes in the rest of v3 I have made the modification here too.